packages/arrakis/arrakis.1.0.0/opam

```diff
@@ -0,0 +1,43 @@
+opam-version: "2.0"
+synopsis: "A RISC-V simulator"
+description: """
+A RISC-V simulator  with an integrated debugger.
+Support full RV32IM instruction set,
+along with pseudo instruction and several environmental calls."""
+maintainer: ["mvalo@proton.me" "gurvan@debauss.art"]
+authors: ["Valeran Maytié" "Gurvan Debaussart"]
+license: "CeCILL-2.1"
+tags: [
+  "RISC-V" "simulator" "emulation" "RV32IM" "education" "debugger" "debug"
+]
+homepage: "https://gitlab.com/valoranM/arrakis"
+doc: "https://gitlab.com/valoranM/arrakis/-/wikis/"
+bug-reports: "https://gitlab.com/valoranM/arrakis/-/issues"
+depends: [
+  "dune" {>= "3.10"}
+  "ocaml" {>= "4.14.0"}
+  "menhir"
```

Comment:
CI suggests that a lower bound for menhir is needed.

My guess is `>= "20180528"` but not sure whether it is the case.

```suggestion
  "menhir" {>= "20180528"}
```

(from https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/9a04a6d1567a3e929efc4338665dfba8391600ca/variant/compilers,4.14,arrakis.1.0.0,lower-bounds)
```
The following actions will be performed:
=== remove 4 packages
  - remove    menhirLib          20230608             [conflicts with menhir]
  - remove    menhirSdk          20230608             [conflicts with menhir]
  - remove    ocaml-syntax-shims 1.0.0
  - remove    uutf               1.0.3                [conflicts with cmdliner]
=== downgrade 10 packages
  - downgrade alcotest           1.7.0 to 1.0.0       [required by arrakis]
  - downgrade cmdliner           1.2.0 to 1.0.3       [required by alcotest]
  - downgrade fmt                0.9.0 to 0.8.6       [required by alcotest]
  - downgrade menhir             20230608 to 20160303 [required by arrakis]
  - downgrade ocamlbuild         0.14.2 to 0.14.0     [required by uuidm]
  - downgrade ocamlfind          1.9.6 to 1.9.1       [required by uuidm]
  - downgrade re                 1.11.0 to 1.7.2      [required by alcotest]
  - downgrade seq                base to 0.2.2        [required by fmt]
  - downgrade stdlib-shims       0.3.0 to 0.1.0       [required by alcotest]
  - downgrade topkg              1.0.7 to 1.0.3       [required by uuidm]
=== recompile 1 package
  - recompile arrakis            1.0.0 (pinned)
=== install 3 packages
  - install   base-bytes         base                 [required by re, uuidm]
  - install   jbuilder           1.0+beta20.1         [required by re]
  - install   uuidm              0.9.6                [required by alcotest]
```

```
#=== ERROR while compiling arrakis.1.0.0 ======================================#
# context              2.2.0~alpha3~dev | linux/x86_64 | ocaml-base-compiler.4.14.1 | pinned(https://gitlab.com/valoranM/arrakis/-/archive/1.0.0/arrakis-1.0.0.tar.gz)
# path                 ~/.opam/4.14/.opam-switch/build/arrakis.1.0.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p arrakis -j 47 @install
# exit-code            1
# env-file             ~/.opam/log/arrakis-7-a97128.env
# output-file          ~/.opam/log/arrakis-7-a97128.out
### output ###
# File "/home/opam/.opam/4.14/lib/re/re.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.14/lib/re/emacs/re.emacs.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.14/lib/re/glob/re.glob.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.14/lib/re/pcre/re.pcre.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.14/lib/re/perl/re.perl.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.14/lib/re/posix/re.posix.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.14/lib/re/str/re.str.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "arrakis/lib/assembler/dune", line 1, characters 0-38:
# 1 | (menhir
# 2 |  (flags -v)
# 3 |  (modules parser))
# (cd _build/.sandbox/04e5d1ef3ace8fc6de98fe304e56da79/default && /home/opam/.opam/4.14/bin/menhir -v arrakis/lib/assembler/parser.mly --base arrakis/lib/assembler/parser --infer-write-query arrakis/lib/assembler/parser__mock.ml.mock)
# /home/opam/.opam/4.14/bin/menhir: unknown option '--infer-write-query'.
# Usage: /home/opam/.opam/4.14/bin/menhir <options> <filenames>
#   --base <basename>           Specifies a base name for the output file(s)
#   --canonical                 Construct a canonical Knuth LR(1) automaton
#   --comment                   Include comments in the generated code
#   --compare-errors <filename> (used twice) Compare two .messages files.
#   --compile-errors <filename> Compile a .messages file to OCaml code.
#   --coq                       Generate a formally verified parser, in Coq
#   --coq-no-complete           Do not generate a proof of completeness
#   --coq-no-actions            Ignore semantic actions in the Coq output
#   --depend                    Invoke ocamldep and display dependencies
#   --dump                      Describe the automaton in <basename>.automaton
#   --echo-errors <filename>    Echo the sentences in a .messages file
#   --error-recovery            (no longer supported)
#   --explain                   Explain conflicts in <basename>.conflicts
#   --external-tokens <module>  Import token type definition from <module>
#   --fixed-exception           Declares Error = Parsing.Parse_error
#   --follow-construction       (undocumented)
#   --graph                     Write grammar's dependency graph to <basename>.dot
#   --infer                     Invoke ocamlc for ahead of time type inference
#   --inspection                Generate the inspection API (requires --table)
#   --interpret                 Interpret the sentences provided on stdin
#   --interpret-show-cst        Show a concrete syntax tree upon acceptance
#   --interpret-error           Interpret one sentence that should end in an error
#   --lalr                      Construct an LALR(1) automaton
#   --list-errors               Produce a list of erroneous inputs
#   --log-automaton <level>     Log information about the automaton
#   --log-code <level>          Log information about the generated code
#   --log-grammar <level>       Log information about the grammar
#   --no-code-inlining          (undocumented)
#   --no-inline                 Ignore the %inline keyword.
#   --no-pager                  (undocumented)
#   --no-prefix                 (undocumented)
#   --no-stdlib                 Do not load the standard library
#   --ocamlc <command>          Specifies how ocamlc should be invoked
#   --ocamldep <command>        Specifies how ocamldep should be invoked
#   --only-preprocess           Print grammar and exit
#   --only-preprocess-u         Print grammar with unit actions and exit
#   --only-preprocess-uu        Print grammar with unit actions & tokens and exit
#   --only-tokens               Generate token type definition only, no code
#   --raw-depend                Invoke ocamldep and echo its raw output
#   --stdlib <directory>        Specify where the standard library lies
#   --strict                    Warnings about the grammar are errors
#   --suggest-comp-flags        Suggest compilation flags for ocaml{c,opt}
#   --suggest-link-flags-byte   Suggest link flags for ocamlc
#   --suggest-link-flags-opt    Suggest link flags for ocamlopt
#   --suggest-menhirLib         Suggest where is MenhirLib
#   --suggest-ocamlfind         Show whether Menhir was installed using ocamlfind
#   --table                     Use the table-based back-end
#   --timings                   Display internal timings
#   --trace                     Include tracing instructions in the generated code
#   --unused-token <token>      Do not warn that <token> is unused
#   --unused-tokens             Do not warn about any unused token
#   --update-errors <filename>  Update auto-comments in a .messages file
#   --version                   Show version number and exit
#   -b <basename>               Synonymous with --base <basename>
#   -lg                         Synonymous with --log-grammar
#   -la                         Synonymous with --log-automaton
#   -lc                         Synonymous with --log-code
#   -t                          Synonymous with --table
#   -v                          Synonymous with --dump --explain
#   -help                       Display this list of options
#   --help                      Display this list of options
```